iec61850serversimulator/src/main/java/com/cgi/iec61850serversimulator/Scheduler.java

```diff
@@ -15,98 +16,106 @@
+        // it.
+        // Calculate relative time for each task
+        // Schedule it, with future put into list
+        this.clearFuture(this.scheduledFutures);
+        for (int i = 0; i < switchingMoments.size(); i++) {
+            SwitchingMoment switchingMoment = switchingMoments.get(i);
+            LocalDateTime currentTime = LocalDateTime.now();
+            // Calculate relative time
+            int relativeTime = this.timeCalculator.calculateSecondsUntil(switchingMoment.getTriggerTime(), currentTime);
+            int relayNr = switchingMoment.getRelayNr();
+            if (switchingMoment.isTriggerAction()) {
+                Runnable runOn = this.onRunnableCreator(relayNr);
+                this.scheduledFutures.add(this.executor.schedule(runOn, relativeTime, TimeUnit.SECONDS));
+                logger.info("Switching Moment for On action created! Relative time: {} ", relativeTime);
+            } else {
+                Runnable runOff = this.offRunnableCreator(relayNr);
+                this.scheduledFutures.add(this.executor.schedule(runOff, relativeTime, TimeUnit.SECONDS));
+                logger.info("Switching Moment for Off action created! Relative time: {} ", relativeTime);
+            }
+        }
+        logger.info("Schedules planned!");
+    }
 
-	}
+    // Runnable to turn on relay light
+    private Runnable onRunnableCreator(final int relayNr) {
+        Runnable onRun = new Runnable() {
+            @Override
+            public void run() {
+                Scheduler.this.device.getRelay(relayNr).setLight(true);
+            }
+        };
+        return onRun;
+    }
 
-	// Runnable to turn on relay light
-	private Runnable onRunnableCreator(final int relayNr) {
-		Runnable onRun = new Runnable() {
-			@Override
-			public void run() {
-				Scheduler.this.device.getRelay(relayNr).setLight(true);
-			}
-		};
-		return onRun;
-	}
+    // Runnable to turn off relay light
+    private Runnable offRunnableCreator(final int relayNr) {
+        Runnable offRun = new Runnable() {
+            @Override
+            public void run() {
+                Scheduler.this.device.getRelay(relayNr).setLight(false);
+            }
+        };
+        return offRun;
+    }
 
-	// Runnable to turn off relay light
-	private Runnable offRunnableCreator(final int relayNr) {
-		Runnable offRun = new Runnable() {
-			@Override
-			public void run() {
-				Scheduler.this.device.getRelay(relayNr).setLight(false);
-			}
-		};
-		return offRun;
-	}
-
-	private void clearFuture(List<ScheduledFuture<?>> scheduledFutures) {
-		if (!scheduledFutures.isEmpty()) {
-			for (int i = 0; i < scheduledFutures.size(); i++) {
-				ScheduledFuture<?> future = scheduledFutures.get(i);
-				if (!future.isDone()) {
-					future.cancel(true);
-				}
-			}
-			scheduledFutures.clear();
-			// To prevent memory leak: Purging executor's queue
-			this.executor.purge();
-		}
-	}
+    private void clearFuture(List<ScheduledFuture<?>> scheduledFutures) {
```

Comment:
Clearer name: clearScheduledFutures (and I would even call it clearScheduledSwitchingMoments).

Reply:
Done, also renamed the variable to scheduledSwitchingMoments to make the variable name also clear.